Added a middle bhop cap (slows players from skimming quicker)

mikeparker opened the pull request:
When you are over it, reduce speed by more. Currently set to 155% speed (620 for scout) and it has a lower pcfactor (0.4 rather than 0.65). This means downtrimping is good for one jump and reduces successive conc skim jumps without reducing the hard cap too much.

AdamWillden commented:
I wonder if 0.3 pcfactor is a little harsh or 155% a little low. I approve of the idea overall though and it's a starting point. I need to get my machine set up to compile & test.

I'v been downloading SDK update and FF for days, I get 1.4Mb at best :(

Just going to link this with #36 as they're related

mikeparker commented:
Without this change, a conc skim as scout, with a bit of turning / air accel between jumps goes:
conc (1400) -> 720 -> 666 -> 640 -> 620 -> 605 -> 596 -> bhop speed (6 jumps above bhop speed)

Similarly, a bhop into a single down trimp follows the same pattern, you get 6 jumps above bhop speed for a single downtrimp:
580 -> 720 (downtrimp) -> 666 -> 640 -> 620 -> 605 -> 596 -> bhop speed (6 jumps above bhop speed)

With this change, instead it goes:
conc (1400) -> 720 -> 608 -> bhop speed (2 jumps above bhop speed)

With pcfactor 0.5 instead of 0.3 it goes:
conc (1400) -> 720 -> 640 -> 600 -> bhop speed (3 jumps above bhop speed)

With pcfactor 0.4:
conc (1400) -> 720 -> 630 -> 600 -> bhop speed (3 jumps above bhop speed)

I guess i'd be happy anywhere between 0.3 and 0.5, I think 0.4 might actually feel the best. 
bear in mind that our old high cap (200%) was subject to the random 0.9 multiplier so it was actually 180%. With my other PR in, that leaves the caps at:
140% soft cap 35% reduction to anything over the limit
155% mid cap 60% reduction to anything over the limit
171% hard cap 100% reduction to anything over the limit

mikeparker commented:
Ping @squeek502 for thoughts

squeek502 commented:
Wanted to test this locally first. Will comment later tonight.

squeek502 commented:
I like it. Doesn't feel like a brick wall and is a nice fairly subtle nerf to skimming.

:+1:

mikeparker commented on game_shared/ff/ff_gamemovement.cpp:
```diff
@@ -36,7 +36,7 @@
 #define SV_TRIMPMULTIPLIER 1.4f
 //static ConVar sv_trimpdownmultiplier("sv_trimpdownmultiplier", "1.2", FCVAR_REPLICATED | FCVAR_CHEAT);
-#define SV_TRIMPDOWNMULTIPLIER 1.2f
+#define SV_TRIMPDOWNMULTIPLIER 1.1f
```
I don't think I meant to commit this line - i was experimenting with reducing the downtrimp amount

squeek502 replied:
Yeah, I'd like to see this reverted before the PR is merged.

mikeparker committed:
Added a middle bhop cap - when you are over it, reduce speed by more. Currently set to 1.55 speed (620 for scout) and it has a lower pcfactor (0.3 rather than 0.65). This means downtrimping is good for one jump and reduces successive conc skim jumps without reducing the hard cap too much

Increase pcfactor_mid from 0.3 to 0.4 to maintain _slightly_ more speed above the mid cap

Revert accidental downtrimp nerf

mikeparker committed:
Fix up some comments and things
